woocommerce/woocommerce-android
# Handle fee selection

Summary
==========
Fixes issue #5696 by connecting the Fee view selection with the Order draft updates submission, also allowing editing mode and removing the selected fee.

Screenshots
==========
| Add fee option available inside the New order view | Creating a new fee | Fee button disabled when the price is getting updated | Fee button available with fee price set | Fee view in editing mode with remove button | Fee view in editing mode with percentage mode also calculated |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| ![Screenshot_20220224_225328](https://user-images.githubusercontent.com/5920403/155640381-9f7fdbdb-5427-4cb8-9d24-3f035ec5c963.png) | ![Screenshot_20220224_225345](https://user-images.githubusercontent.com/5920403/155640500-8edf7783-06b0-4982-8922-ace55cb39b60.png) | ![Screenshot_20220224_225403](https://user-images.githubusercontent.com/5920403/155640514-d8c4691e-a1d7-4ef2-a039-34a89a7f3c81.png) | ![Screenshot_20220224_225414](https://user-images.githubusercontent.com/5920403/155640518-dc04eed5-62d6-4e7a-8cf9-63aff0461a6e.png) | ![Screenshot_20220224_225422](https://user-images.githubusercontent.com/5920403/155640525-45731965-52ca-4cf9-8617-d2fc0ec9921d.png) | ![Screenshot_20220224_225431](https://user-images.githubusercontent.com/5920403/155640534-7c5de723-bd52-4e03-8a38-c73780a4aa86.png) |

How to Test
==========
1. Make sure the feature flag `ORDER_CREATION_M2` is enabled.
2. Open order creation form.
3. Add a product.
4. Notice the button "Add fee" on the payment section and click on it.
5. Enter an amount and hit `done`.
6. Notice that the New order view is back, loading the newly entered fee price into the Order total, and the fee button is disabled.
7. After the loading is finished, the Fee button should be enabled back again, the text is now only `Fees` instead of `Add fee`, and the price is correctly calculated.
8. When hitting the `Fee` button, the Fee view should now display the `remove fee` button, both $ and % edit fields should be displaying values based on the current fee, and editing or removing any of them should work as expected.


Update release notes:
